### PR TITLE
feat(ui): Introduce PureBlack switch on dark theme

### PR DESCRIPTION
**Pull request by dtomvan**
This works freely on any theme that has dark colors (as of now: AMOLED,
Dark, System while in dark mode) while using any "base theme".

NOTE: I don\'t know if me and @25huizengek1 are entirely happy on the behaviour
but man, this theme stuff sure is difficult isn\'t it? Also my first time using
Kotlin for Android apps with Compose, so I hope the code looks good enough. I
don't know which formatter I should be using, so I used none but tried to indent
cleanly by myselves (looks like 4 spaces across the board).

@jojobozo, does this look good to you?

Resolves: #142

### CODE REVIEW

**Comment by dtomvan** [comment]
Wow, that just took me 45 minutes to figure everything out and writing these meesly 30 lines...

**Comment by dtomvan** [comment]
@jojobozo, here's a debug APK file for your convenience: https://www.mediafire.com/file/a49h5jfrccvza4u/app-debug.apk/file

Could you please install this, try out the various theme selection options and get back to me with how you like it?

File: app/src/main/kotlin/app/vitune/android/preferences/AppearancePreferences.kt [hunk 1]
```diff
@@ -9,6 +9,7 @@ import app.vitune.core.ui.enums.ThumbnailRoundness
 object AppearancePreferences : GlobalPreferencesHolder() {
     var colorPaletteName by enum(ColorPaletteName.Dynamic)
     var colorPaletteMode by enum(ColorPaletteMode.System)
+    var usePureBlack by boolean(false)
```
Comment (25huizengek1) [inline app/src/main/kotlin/app/vitune/android/preferences/AppearancePreferences.kt:12]: Not backwards compatible? Or am I missing something here?

File: core/ui/src/main/kotlin/app/vitune/core/ui/enums/ColorPaletteName.kt [hunk 2]
```diff
@@ -4,6 +4,5 @@ enum class ColorPaletteName(val isDynamic: Boolean) {
     Default(isDynamic = false),
     Dynamic(isDynamic = true),
     MaterialYou(isDynamic = true),
-    PureBlack(isDynamic = false),
```
Comment (25huizengek1) [inline core/ui/src/main/kotlin/app/vitune/core/ui/enums/ColorPaletteName.kt:7]: Yup, removing this could even cause a crash I guess.

**Comment by 25huizengek1** [comment]
@dtomvan Thank you for your implementation though, appreciate it.

I suggest you take a look at how I 'fixed' this: https://github.com/25huizengek1/ViTune/commit/887cd9f7e0cfe88b4b51f083888904791d886922#diff-7aac4d03c3f31e309738038c4c324763569934908f872293a796e769b974add2R16

**Comment by dtomvan** [comment]
Hahah, I forgot these config classes have representations in some "GlobalPreferencesHolder" so once you alter these classes you are altering the config format...

**Comment by dtomvan** [comment]
Also, are "missing fields" in previous config files not allowed?! How is that sensible?

File: app/src/main/kotlin/app/vitune/android/preferences/AppearancePreferences.kt (hunk 1 above)
Comment (dtomvan) [reply app/src/main/kotlin/app/vitune/android/preferences/AppearancePreferences.kt:12]: So I use a `private val` to "cover up" usePureBlack and revert to the old "multi-purpose-enum" approach from before?

File: core/ui/src/main/kotlin/app/vitune/core/ui/enums/ColorPaletteName.kt (hunk 2 above)
Comment (dtomvan) [reply core/ui/src/main/kotlin/app/vitune/core/ui/enums/ColorPaletteName.kt]: Re-added it in the fixup. Let's autosquash later

**Comment by 25huizengek1** [comment]
ffa958c9b9bf7776359c91222ad74bdaac08e384 Because of the upgraded dependencies in master

**Comment by dtomvan** [comment]
I'll close this as we have come up with a better way to do this: @25huizengek1 will come up with a PR/commit shortly.

**Pull request closed without merging**